test/E2E.test.ts

```diff
@@ -0,0 +1,329 @@
+
+const { formatBytes32String } = ethers.utils;
+const {
+  STAGING_LINK_TOKEN,
+  STAGING_LINK_USD_PRICE_FEED,
+  STAGING_PAYMENT_USD_PRICE_FEED,
+  STAGING_SWAP_ROUTER,
+} = process.env as ProcessEnv;
+
+const TopUpParams = {
+  MIN_WITHDRAW_AMT_PAYMENT_TOKEN: ethers.utils.parseEther("1"),
+  MAX_DEPOSIT_AMT_PAYMENT_TOKEN: ethers.utils.parseEther("3"),
+  THRESHOLD_LINK: ethers.utils.parseEther("7"),
+};
+
+const VestParams = {
+  TOTAL_AMOUNT_VESTING_PLAN: ethers.utils.parseEther("100"),
+  DURATION_IN_SEC_VESTING_PLAN: 1,
+  CLIFF_PERIOD_IN_SEC: 0,
+};
+
+const UpkeepParams = {
+  NAME: "test123",
+  ADMIN_EMAIL: "test@example.com",
+  GAS_LIMIT: 500000,
+  INITIAL_FUNDING: ethers.utils.parseEther("8.0"),
+  CHECK_DATA: "0x",
+  OFFCHAIN_CONFIG: "0x",
+};
+
+const MIN_UPKEEP_SPEND = BigNumber.from("1000");
+const AUTO_APPROVE_CONFIG_TYPE = {
+  NONE: 0,
+  WHITELIST: 1,
+  ALLOW_ALL: 2,
+};
+const AUTO_APPROVE_MAX_ALLOWED = BigNumber.from(20);
+const KEEPER_REGISTRY_LOGIC = "0x96d2971f181bf01f7b61f254e0ded20bb1657e7b";
+const FAKE_VOW_ADDRESS = "0xA950524441892A31ebddF91d3cEEFa04Bf454466";
+const NONFUNGIBLE_POSITION_MANAGER =
+  "0xC36442b4a4522E871399CD717aBDD847Ab11FE88"; // Its the same address on every network
+
+const DSS_VEST_CAP = ethers.utils.parseEther("100"); // Maximum per-second issuance token rate
+
+function getRegistrySigners() {
+  const signers: Wallet[] = [];
+  const signer1 = new ethers.Wallet(
+    "0x7777777000000000000000000000000000000000000000000000000000000001"
+  ).connect(ethers.provider);
+  const signer2 = new ethers.Wallet(
+    "0x7777777000000000000000000000000000000000000000000000000000000002"
+  ).connect(ethers.provider);
+  const signer3 = new ethers.Wallet(
+    "0x7777777000000000000000000000000000000000000000000000000000000003"
+  ).connect(ethers.provider);
+  const signer4 = new ethers.Wallet(
+    "0x7777777000000000000000000000000000000000000000000000000000000004"
+  ).connect(ethers.provider);
+  const signer5 = new ethers.Wallet(
+    "0x7777777000000000000000000000000000000000000000000000000000000005"
+  ).connect(ethers.provider);
+
+  signers.push(signer1, signer2, signer3, signer4, signer5);
+
+  return signers;
+}
+
+describe("DssCronKeeper KeeperRegistry", function () {
+  let keeper: DssCronKeeper;
+  let sequencer: Sequencer;
+  let job: SampleJob;
+  let registry: KeeperRegistry20;
+  let registrar: KeeperRegistrar20;
+  let linkToken: LinkTokenMock;
+  let upkeepId: string;
+  let vestId: BigNumber;
+  let owner: SignerWithAddress;
+  let registrySigners: Wallet[];
+  let dssVest: DssVestMintable;
+  let topUp: DssVestTopUp;
+  let paymentToken: ERC20PresetMinterPauser;
+
+  const f = 1;
+  if (
+    !STAGING_LINK_TOKEN ||
+    !STAGING_LINK_USD_PRICE_FEED ||
+    !STAGING_PAYMENT_USD_PRICE_FEED ||
+    !STAGING_SWAP_ROUTER
+  ) {
+    throw new Error("Missing required env variables!");
+  }
+
+  before(async function () {
```

Comment:
What do you think about moving each step of the setup process to a separate function in this file?

Reply:
There are some steps left: setup sequencer, sample job, registry, create and fund pool. 

Also, some steps could be combined: deploy registry + registrar + config, create + fund pool.

Further, when function names are descriptive code comments are not needed anymore.

The goal is to be very easy to read and understand.